### DISCUSSION TITLE
WSL: Mount linux folder in kubernetes

### DISCUSSION THREAD

**Discussion by benzman81** [rancher-sandbox/rancher-desktop#2555] [Q&A] [answered]
Hi,
I couldnt find any answer to this. How can I mount a folder in WSL linux using kubernetes hostPath? If I use the linux path `/home/myuser/foo` then an empty folder with this path is created in `\\wsl$\rancher-desktop` but not the linux path from WSL is mounted. I know that I have acces to /mnt/c/ for my windows system, but this doesnt help here. Any hints?
Regards,
Markus

**Comment by adamkpickering** [comment, marked as answer]
My understanding of the issue is below. @mook-as if you have anything to add, it would help!

There are three entities we need to be aware of:
1. the host machine
2. the RD VM, which is where k8s, dockerd and/or containerd run
3. the user VM that you are trying to mount a directory from

As you have noticed, both the RD VM and the user VM have access to the C drive on the host machine via /mnt/c/. I'm not aware of any other mounts, but I could be missing something. A simple solution, then, could be working in the /mnt/c directory. That way, any files created in that folder in the user VM would get passed through to the host machine, and then to the RD VM. So you could use the same path for your `hostPath` config as you are working with in the user VM.

I don't have a better answer than that. You could also try messing around with symlinks to /mnt/c in the user VM if you want your stuff in a different directory. Does that answer your question?

**Reply by mook-as** [reply]
The relevant difference is that there's actually one VM, but two containers (the RD one, and the other distro) running on top of it. (RD then runs nested containers via dockerd/containerd.)

`/mnt/wsl` is a shard mount point across all distributions; that is, it's a mount point that is mounted into each distribution.  This means that you can achieve host path mounting from a different distribution by:
1. From the target distro, bind mount `/home/myuser/foo` into `/mnt/wsl/foo`.
2. In your hostPath configuration, tell it to use `/mnt/wsl/foo`.

This is in fact what we do in [nerdctl-stub](https://github.com/rancher-sandbox/rancher-desktop/tree/main/src/go/nerdctl-stub) to handle volume mounts when running `nerdctl` from a WSL distribution.  We also do this (backwards) to expose `docker.sock` to the other distributions when using dockerd.

(Note that the mount prefix is [configurable per distribution](https://docs.microsoft.com/en-us/windows/wsl/wsl-config#automount-settings).)

**Reply by benzman81** [reply]
Thanks. Now I have a better understanding of how it work and what I can do to mount my folder.
